File: selection.py
def selectionSort(data,length):
    firstUnsorted=0
    while(firstUnsorted<length):
        indexOfSmallest=firstUnsorted
        index=firstUnsorted+1
        for i in range(index,length):
            if(data[i]<=data[indexOfSmallest]):
                indexOfSmallest=i
        data[firstUnsorted],data[indexOfSmallest]=data[indexOfSmallest],data[firstUnsorted]
        firstUnsorted+=1

File: selection.c
selectionSort(int *data,int length){
	int firstUnsorted;

	firstUnsorted=0;

	while(firstUnsorted<length) {
		int indexOfSmallest;
		int index;
		int temp;		

		indexOfSmallest=firstUnsorted;
		index=firstUnsorted+1;

		int i;

		for(i=index;i<length;i++) {
			if(data[i]<=data[indexOfSmallest]) indexOfSmallest=i;
		}
		temp=data[firstUnsorted];
		data[firstUnsorted]=data[indexOfSmallest];
		data[indexOfSmallest]=temp;
		firstUnsorted=firstUnsorted+1;
	}
}
